backend/flowpilot/input.go

```diff
@@ -239,7 +242,7 @@ func (i *DefaultInput) validate(stateName StateName, inputData ReadOnlyActionInp
 		}
 	}
 
-	if i.dataType == EmailType {
+	if i.dataType == EmailType && !isRequired && !hasEmptyOrNilValue {
```

Comment:
The value should be matched against the regex when `isRequired` is `true` as well.